plugins/callback/opentelemetry.py

```diff
@@ -0,0 +1,374 @@
+    from opentelemetry import trace
+    from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExporter
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+    from opentelemetry.trace.status import Status, StatusCode
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import (
+        ConsoleSpanExporter,
+        SimpleSpanProcessor,
+        BatchSpanProcessor
+    )
+    from opentelemetry.util._time import _time_ns
+    HAS_OTEL = True
+except ImportError:
+    HAS_OTEL = False
+
+try:
+    from collections import OrderedDict
+    HAS_ORDERED_DICT = True
+except ImportError:
+    try:
+        from ordereddict import OrderedDict
+        HAS_ORDERED_DICT = True
+    except ImportError:
+        HAS_ORDERED_DICT = False
+
+
+class CallbackModule(CallbackBase):
+    """
+    This callback creates distributed traces.
+    This plugin makes use of the following environment variables:
+        SPAN_ID (optional): A valid span identifier to be used as the parent span.
+        TRACE_ID (optional): A valid trace identifier to be used as the parent trace.
+        INCLUDE_SETUP_TASKS (optional): Should the setup tasks be includedt
+                                     Default: True
+        HIDE_TASK_ARGUMENTS (optional): Hide the arguments for a task
+                                     Default: False
+        OTEL_SERVICE_NAME (optional): The service name
+                                     Default: ansible
+        OTEL_EXPORTER (optional): Use the OTEL exporter and its environment variables. https://opentelemetry-python.readthedocs.io/en/latest/exporter/otlp/otlp.html
+                                     Default: False
+        OTEL_EXPORTER_INSECURE (optional): Insecure connection
+                                     Default: False
+    Requires:
+        opentelemetry-api
+        opentelemetry-exporter-otlp
+        opentelemetry-sdk
+    """
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'community.general.opentelemetry'
+    CALLBACK_NEEDS_ENABLED = True
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+
+        self._include_setup_tasks = os.getenv('INCLUDE_SETUP_TASKS', 'True').lower()
+        self._hide_task_arguments = os.getenv('HIDE_TASK_ARGUMENTS', 'False').lower()
+        self._service = os.getenv('OTEL_SERVICE_NAME', 'ansible').lower()
+        self._otel_exporter = os.getenv('OTEL_EXPORTER', 'False').lower() == 'true'
+        self._insecure_otel_exporter = os.getenv('OTEL_EXPORTER_INSECURE', 'False').lower() == 'true'
+        self._span_id = os.getenv('SPAN_ID', None)
+        self._trace_id = os.getenv('TRACE_ID', None)
+        self._playbook_path = None
+        self._playbook_name = None
+        self._play_name = None
+        self._task_data = None
+        self.errors = 0
+        self.disabled = False
+
+        if not HAS_OTEL:
+            self.disabled = True
```

Comment:
Instead of setting some flag, you should fail (raise an appropriate exception).

Reply:
I saw this particular behaviour in some other callback plugins. Therefore, I'm not sure if an exception should be raised in case the required libraries were not installed but disabling this plugin as the less disruptive solution.
What do you think?

Reply:
I've asked around a bit for this one, disabling is probably the better approach. So please keep it as-is.

Reply:
I've tested with your suggestion in https://github.com/ansible-collections/community.general/pull/3091#discussion_r703037883 and it works like a charm:


```bash
$ pip3 uninstall opentelemetry-sdk
...
$ ANSIBLE_OPENTELEMETRY_CONSOLE_OUTPUT=true \
   ansible-playbook -i hosts test.yml
[WARNING]: Skipping callback 'opentelemetry', unable to load due to: The `opentelemetry-api`, `opentelemetry-exporter-otlp` or `opentelemetry-sdk` must be installed to use this plugin
...
PLAY RECAP ***************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=6    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```